packages/Webkul/Core/src/Helpers/Laravel5Helper.php

```diff
@@ -354,49 +317,74 @@ private function createDownloadableLink(int $productId): void
     }
 
     /**
-     * @param int   $productId
-     * @param array $attributeValues
-     *
-     * @return void
+     * @param Product $product
+     * @param array   $attributeValues
      */
-    private function createAttributeValues(int $productId, array $attributeValues = []): void
+    private function createAttributeValues(Product $product, array $attributeValues = []): void
     {
         $I = $this;
 
-        $productAttributeValues = [
-            'sku',
-            'url_key',
-            'tax_category_id',
-            'price',
-            'cost',
-            'name',
-            'new',
-            'visible_individually',
-            'featured',
-            'status',
-            'guest_checkout',
-            'short_description',
-            'description',
-            'meta_title',
-            'meta_keywords',
-            'meta_description',
-            'weight',
+        $faker = Factory::create();
+
+        $brand = Attribute::where(['code' => 'brand'])->first(); // usually 25
+
+        if (! AttributeOption::where(['attribute_id' => $brand->id])->exists()) {
+            AttributeOption::create([
+                'admin_name'   => 'Webkul Demo Brand (c) 2020',
+                'attribute_id' => $brand->id,
+            ]);
+
+        }
+
+        /** @var array $defaultAttributeValues
+         * Some defaults that should apply to all generated products.
+         * By defaults products will be generated as saleable.
+         * If you do not want this, this defaults can be overriden by $attributeValues.
+         */
+        $defaultAttributeValues = [
+            'name'                 => $faker->word,
+            'description'          => $faker->sentence,
+            'short_description'    => $faker->sentence,
+            'sku'                  => $faker->word,
+            'url_key'              => $faker->slug,
+            'status'               => true,
+            'visible_individually' => true,
+            'special_price_from'   => null,
+            'special_price_to'     => null,
+            'special_price'        => null,
+            'price'                => '1.00',
+            'weight'               => '1.00', // necessary for shipping
+            'brand'                => AttributeOption::firstWhere('attribute_id', $brand->id)->id,
         ];
 
-        foreach ($productAttributeValues as $attribute) {
-            $data = ['product_id' => $productId];
+        $attributeValues = array_merge($defaultAttributeValues, $attributeValues);
 
-            if (array_key_exists($attribute, $attributeValues)) {
-                $fieldName = self::getAttributeFieldName($attribute);
+        /** @var array $possibleAttributeValues list of the possible attributes a product can have */
+        $possibleAttributeValues = Attribute::all()->pluck('code')->toArray();
```

Comment:
Wouldn't it be better to get attribute id, code and type at once?
DB::table('attributes')->select('id', 'code', 'type')->get()->toArray();

Reply:
Yes, this would be a good microoptimization. Will do this soon, please not merge yet, thanks :)

Reply:
Nice.

Reply:
I have optimized as you suggested and reopen this issue now. It is ready to be reviewed.